File: 1615-maximal-network-rank/1615-maximal-network-rank.cpp
class Solution {
public:
    int maximalNetworkRank(int n, vector<vector<int>>& roads) {
        unordered_map<int,unordered_map<int,int>>mp;
        unordered_map<int,int>smp;
        int ma=0;
        int hp=0;
        for(int i=0;i<roads.size();i++)
        {
            mp[roads[i][0]][roads[i][1]]=1;
            mp[roads[i][1]][roads[i][0]]=1;
            smp[roads[i][0]]++;
            smp[roads[i][1]]++;
        }
        int res=0;
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<n;j++)
            {
                if(j==i)
                    continue;
                int a=smp[i]+smp[j];
                if(mp[i][j])
                {
                    res=max(res,a-1);
                }
                else
                    res=max(res,a);
                
            }
        }
        return res;
        
        
        
        
    }
};

File: 1615-maximal-network-rank/1615-maximal-network-rank.java
class Solution {
    public int maximalNetworkRank(int n, int[][] roads) {
        ArrayList<HashSet<Integer>>graph=new ArrayList<HashSet<Integer>>();
        for(int i=0;i<n;i++)
        {
            graph.add(new HashSet<Integer>());
        }
        for(int i=0;i<roads.length;i++)
        {
            graph.get(roads[i][0]).add(roads[i][1]);
            graph.get(roads[i][1]).add(roads[i][0]);
        }
        int ma=0;
        for(int i=0;i<graph.size();i++)
        {
            for(int j=i+1;j<graph.size();j++)
            {
                int flag=0;
                if(graph.get(i).contains(j))
                    flag=1;
                ma=Math.max(ma,graph.get(i).size()+graph.get(j).size()-flag);
            }
        }
        return ma;
        
    }
}